### PR TITLE
Add token XOR and RLike,Make some keyword can be used as column name in sql

### LINKED ISSUE #88: [Bug]: `SELECT id FROM move_title WHERE id > ('1' | '2')` can not be parse
**Describe the bug**
`SELECT id FROM move_title WHERE id > ('1' | '2')` can not be parse

### PR DESCRIPTION
**Pull request by qidi1**
Close https://github.com/oceanbase/sql-lifecycle-management/issues/88

### Task Description

* add token XOR and RLike
* repair bug `SELECT id FROM move_title WHERE id > ('1' | '2')` cant be resovle  
* make some keyword can be used as column name in sql
     now support: CAST END ESCAPE FOR FROM GROUP IF IN INTO IS ON OR USE WITH ENGINE
      If you want to add more keywords, please refer to the `p_keyword_can_use` method in the parser/mysql_parser/parser.py file.

### Passed Regressions
* Unittest

### Upgrade Compatibility
None

### CODE REVIEW

File: src/parser/mysql_parser/parser.py [hunk 1]
```diff
@@ -1107,6 +1119,26 @@ def p_non_reserved(p):
...
+                        | GROUP
+                        | IF
+                        | IN
+                        | INTO
+                        | IS
+                        | ON
+                        | OR
+                        | USE
+                        | WITH
+                        | ENGINE
```
Comment (luliwjc) [inline src/parser/mysql_parser/parser.py]: top is also can use?

**Comment by luliwjc** [comment]
repair bug SELECT id FROM move_title WHERE id > ('1' | '2') cant be resovle
Please add a issue and link to it

File: src/parser/mysql_parser/parser.py (hunk 1 above)
Comment (qidi1) [reply src/parser/mysql_parser/parser.py]: ADD

**Comment by Ifffff** [comment]
LGTM

**Pull request merged**